bgpd/bgp_route.c

```diff
@@ -1752,18 +1752,22 @@ int subgroup_announce_check(struct bgp_node *rn, struct bgp_path_info *pi,
 	 * if
 	 * the peer (group) is configured to receive link-local nexthop
 	 * unchanged
-	 * and it is available in the prefix OR we're not reflecting the route
-	 * and
+	 * and it is available in the prefix OR we're not reflecting the route,
+	 * link-local nexthop address is valid and
 	 * the peer (group) to whom we're going to announce is on a shared
 	 * network
 	 * and this is either a self-originated route or the peer is EBGP.
+	 * By checking if nexthop LL address is valid we are sure that
+	 * we do not announce LL address as `::`.
 	 */
 	if (NEXTHOP_IS_V6) {
 		attr->mp_nexthop_len = BGP_ATTR_NHLEN_IPV6_GLOBAL;
 		if ((CHECK_FLAG(peer->af_flags[afi][safi],
 				PEER_FLAG_NEXTHOP_LOCAL_UNCHANGED)
 		     && IN6_IS_ADDR_LINKLOCAL(&attr->mp_nexthop_local))
-		    || (!reflect && peer->shared_network
+		    || (!reflect
+			&& IN6_IS_ADDR_LINKLOCAL(&peer->nexthop.v6_local)
```

Comment:
Can you please point me where this is getting assigned?
Also, when the prefix is updated with both nexthops and the LL is removed, do we send an update with just the global nexthop or a withdrawal if now there is no nexthop? Can you please check?

Reply:
Please also provide some simple UT output.

Reply:
>Can you please point me where this is getting assigned?

you should take a look at `bgp_zebra_nexthop_set()`

>Also, when the prefix is updated with both nexthops and the LL is removed, do we send an update with just the global nexthop or a withdrawal if now there is no nexthop? Can you please check?

I'll check this situation.

>Please also provide some simple UT output.

You mean print something in user terminal?